packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -166,7 +166,13 @@ export abstract class TelemetryLogger implements ITelemetryLogger {
      * @param error - optional error object to log
      */
     public sendErrorEvent(event: ITelemetryErrorEvent, error?: any) {
-        this.sendTelemetryEventCore({ ...event, category: "error" }, error);
+        this.sendTelemetryEventCore({
+            // ensure the error field has some value,
+            // this can and will be overridden by event, or error
```

Comment:
Could be wrong but I don't think this works as you expect. I think prepareErrorObject clobbers event.error no matter what.

Reply:
not if error is undefined, if there is any error i'm fine with this being clobbered. there are tests for this

Reply:
Ahhh I didn't see that `prepareErrorObject` isn't called if the error is undefined.  And I saw you wrote tests for these cases but was confused by how the tests are written that they don't actually compare values but just counts of keys. Not sure they would actually fail if there were certain kinds of bugs here.